ui/components/DataTable.tsx

```diff
@@ -161,31 +161,27 @@ function filterText(
 
   return _.filter(rows, (row) => {
     let matches = false;
-    for (const colName in row) {
-      const value = row[colName];
 
-      const field = _.find(fields, (f) => {
-        if (typeof f.value === "string") {
-          return f.value === colName;
+    fields.forEach((field) => {
+      if (field.textSearchable) {
```

Comment:
You could try adding

```
if (!field.textSearchable) {
  return;
}
```

at the beginning to remove this large `if` wrapping the whole code. It _should_ make `forEach` move to the next element, but please check it. 🙂